xaitk_saliency/impls/vis_sal_classifier/occlusion_scoring.py

```diff
@@ -0,0 +1,51 @@
+from xaitk_saliency import ImageClassifierSaliencyMapGenerator
+from xaitk_saliency.utils.masking import weight_regions_by_scalar
+
+import numpy as np
+
+
+class OcclusionScoring (ImageClassifierSaliencyMapGenerator):
+    """
+    This saliency implementation transforms black-box
+    image classification scores into saliency heatmaps. This should
+    require a sequence of per-class confidences predicted on the
+    reference image, a number of per-class confidences as predicted
+    on perturbed images, as well as the masks of the reference image
+    perturbations (as would be output from a
+    `PerturbImage` implementation.
+    """
+
+    def generate(
+            self,
+            image_conf: np.ndarray,
+            perturbed_conf: np.ndarray,
+            perturbed_masks: np.ndarray
+    ) -> np.ndarray:
+
+        if not (perturbed_masks[0][0][0] in [0, 1]):
+            raise ValueError("Image perturbation mask must be of",
+                             "type integer and binary valued.")
+
+        try:
+            assert len(image_conf) == len(perturbed_conf[0])
+        except AssertionError:
+            raise ValueError("Number of classses in original image and",
+                             " perturbed image do not match.")
```

Comment:
Try-catching an assert is more work than 
```suggestion
        if len(image_conf) != len(perturbed_conf[0]):
            raise ValueError("Number of classses in original image and",
                             " perturbed image do not match.")
```

Reply:
Second observation: We can move these pretty standard checks into the interface and move to a template method pattern w.r.t. the abstract method. This can be a separate PR.

Reply:
Good idea! Sounds good, I will start an issue on making standard checks abstract methods.

Reply:
I already created an issue for the standard checks / template method.